libs/vscode/nx-project-view/src/lib/views/nx-project-util.ts

```diff
@@ -1,23 +1,88 @@
-import path = require('node:path');
+import * as path from 'node:path';
+import * as win32Path from 'path/win32';
+import * as posixPath from 'path/posix';
 
 export function isDefined<T>(val?: T): val is T {
   return !!val;
 }
 
-export const PathHelper = {
+type Entries<T> = {
+  [K in keyof T]: [K, T[K]];
+}[keyof T][];
+
+export function objectEntries<T extends object>(val: T) {
+  return Object.entries(val) as Entries<T>;
+}
+
+export class PathHelper {
+  constructor(private pathApi: typeof path = path) {}
+
+  detailedDirs(val: string) {
```

Comment:
This is quite the function... I get the pain of working with paths in node though. Still, would be great if you could provide typings here. 
Just to make it readable.